### ISSUE TITLE
ci: enforce code style with Black

### ISSUE THREAD

**Issue by uniqueg** [elixir-cloud-aai/foca#237]
Use [Black](https://black.readthedocs.io/en/stable/) to auto-format code and include check in CI workflow.

**Comment by JaeAeich** [comment]
Just a suggestion but I see there are 3 issues for the same, using `pylint`, `black` and `flake8-docstring` would just increase the number of deps. If we are just here to enfoce certain coding style rules why not do it with one dep ([ruff](https://docs.astral.sh/ruff/rules/#pyflakes-f)), checkout the config in [TESK (main branch)](https://github.com/elixir-cloud-aai/TESK/blob/0b81865fc321ae8ad1ae16c04652b81cc51edc26/pyproject.toml#L70C1-L84C29).

**Comment by uniqueg** [comment]
Yup, was thinking the same :+1: But they are still three separate issues. Doesn't mean that we can't close them all with the same PR ;-)
